crypto/merklekeystore/keystore_test.go

```diff
@@ -17,176 +17,185 @@
 package merklekeystore
 
 import (
+	"context"
 	"crypto/rand"
+	"database/sql"
+	"fmt"
+	uuid "github.com/satori/go.uuid"
 	"math"
 	"testing"
 
 	"github.com/algorand/go-algorand/crypto"
 	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/util/db"
 	"github.com/stretchr/testify/require"
 )
 
+// Is this test even needed? What is the purpose?
 func TestSignerCreation(t *testing.T) {
 	a := require.New(t)
+	var err error
 
 	h := genHashableForTest()
 	for i := uint64(1); i < 20; i++ {
-		signer, err := New(i, i+1, 1, crypto.Ed25519Type)
-		a.NoError(err)
+		signer := generateTestSigner(i, i+1, 1, a)
+		defer signer.keyStore.store.Close()
 		_, err = signer.Sign(h, i)
 		a.NoError(err)
 	}
 
-	signer, err := New(0, 0, 1, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(0, len(signer.SignatureAlgorithms))
+	signer := generateTestSigner(0, 0, 1, a)
```

Comment:
maybe we should put this "test_single_limit" in a function. and there will be much less defers

Reply:
Which part exactly are you talking about?